pkg/queue/mem_bounded_queue.go

```diff
@@ -0,0 +1,127 @@
+package queue
+
+import (
+	"sync"
+	"time"
+
+	"github.com/uber/jaeger-lib/metrics"
+)
+
+type Sizeable interface {
```

Comment:
Is there a core API to get the size? Could we use that instead? 

https://golang.org/src/go/types/sizes.go